Voting (#71)
cncf/cnf-wg: charter.md

See [CONTRIBUTING](CONTRIBUTING.md) documentation for more information how to get involved.
-->

## Elections

> Chairs and tech leads will be elected for one year terms and may run for reelection. 

Reviewer: It might be more clear to refer to these as Co-Chairs and not Chairs.

Author: Thanks for the catch!

> Chairs and tech leads will be elected for one year terms and may run for reelection. 

Author: ```suggestion
Co-chairs and tech leads will be elected for one year terms and may run for reelection. 
```

The CNF WG employs "organization voting" to ensure no single organization can dominate the project.

> Individuals not associated with or employed by a company or organization are allowed one organization vote. Each company or organization (regardless of the number of maintainers associated with or employed by that company/organization) receives one organization vote.

Reviewer 2: Per comment below, this leads to very unbalanced representation.

Reviewer 3: This is about voting. See new PR for representation in the elected roles https://github.com/cncf/cnf-wg/pull/75

In other words, if two contributors are employed by Company X, two by Company Y, two by Company Z, and one maintainer is an un-affiliated individual, a total of four "organization votes" are possible; one for X, one for Y, one for Z, and one for the un-affiliated individual.

> 

Author: ```suggestion

For the purpose of this document, “Related Company” shall mean any entity (Company-A) which controls or is controlled by another entity (Company-B) or which, together, is under the common control of a third party (Company-C), in each case where such control results from ownership, either directly or indirectly, of more than fifty percent of the voting securities or membership interests of the entity in question; and “Related Companies” are entities that are each a Related Company as described above.

```

> Each organization listed in the interested parties section has one vote. Any contributor from an organization may cast the vote for that organization. Each organization can cast one vote for a co-chair candidate in each of the communities (Kubernetes (K8s) Community, Service Provider (SP) Community, and CNF Developer (Dev) Community), three total votes. Each organization can vote for as many tech leads as they see fit.

Reviewer 4: We may want to better define "organization". It gets complicated when companies have subsidiaries or op-co's. ONAP had to deal with it a few years back and this is what we ended up with:
https://wiki.onap.org/display/DW/ONAP+Technical+Community+Document#ONAPTechnicalCommunityDocument-4.4.1.4SubcommitteeVoterEligibility

Also, are there any rules for adding an organization to the "interested parties" list? Or do we rely on the common sense of whoever reviews the PR for adding an organization?

Reviewer 2: Also, how do you find out what the organisation-approved vote is?  And are all organisations created equal?

I know why this is in there - to control the possible risk of being overwhelmed by numbers from a specific organisation - but it's not really an open-source-centric model of control being in the hands of participating individuals (e.g committers, as with the OIF).

Reviewer 3: @rannyh for the "interested parties" list common sense when reviewing the PR is used.

Author: @rannyh Thanks for the reference! I've added the wording to make it more clear
I've also added a comment about the interested parties to the PR approval discussion https://github.com/cncf/cnf-wg/discussions/74#discussioncomment-405553

> The chair nominee from each community with the most votes will be elected. Tech leads must win at least 60% of the votes cast to be elected.

Reviewer 5: Open Infra and OPNFV uses the [Ranked voting system](https://en.wikipedia.org/wiki/Ranked_voting) which in my opinion it's a good option. Shouldn't be better to use it here?

Author: Let's discuss this in the meeting on Monday. I'm not opposed to it.

Author: Thanks, just updated to Ranked Choice Voting

> The chair nominee from each community with the most votes will be elected. Tech leads must win at least 60% of the votes cast to be elected.

Reviewer 2: How do we define which community a candidate is in?  How do we define who should be voting for someone in a community?

This may all be a bit too much worrying if the chairs are basically just organisers, mind.

Author: I would expect candidates to self identify and police. If the voters feels the candidate isn't representing the right community, then they can choose against electing them.

In other words, if two contributors are employed by Company X, two by Company Y, two by Company Z, and one maintainer is an un-affiliated individual, a total of four "organization votes" are possible; one for X, one for Y, one for Z, and one for the un-affiliated individual.

For the purpose of this document, “Company” shall mean any entity (Company-A) which controls or is controlled by another entity (Company-B) or which, together, is under the common control of a third party (Company-C), in each case where such control results from ownership, either directly or indirectly, of more than fifty percent of the voting securities or membership interests of the entity in question; and one “Company” is all of the entities that are each described above.

Each organization listed in the interested parties section has one vote. Any contributor from an organization may cast the vote for that organization. Each organization can cast one vote for a co-chair candidate in each of the communities (Kubernetes (K8s) Community, Service Provider (SP) Community, and CNF Developer (Dev) Community), three total votes. Each organization can vote for as many tech leads as they see fit.

> The chair nominee from each community with the most votes will be elected. Tech leads must win at least 60% of the votes cast to be elected.

Author: ```suggestion
The chair for each community will be elected using [Ranked Choice Voting](https://en.wikipedia.org/wiki/Ranked_voting). Tech leads must win at least 60% of the votes cast to be elected.
```
